posthog/queries/query_date_range.py

```diff
@@ -209,6 +209,9 @@ def should_round(self):
         if self._should_round is not None:
             return self._should_round
 
+        if self._filter.use_explicit_dates:
```

Comment:
I'm very confused by this `should_round` function, @EDsCODE .

What's this logic for?

```
round_interval = self.time_difference.total_seconds() >= TIME_IN_SECONDS[self._filter.interval] * 2
```

seems to be unnecessarily adding a `toStartofDay` to trend event filters in all regular queries with > 2 days time range

Unsure if I should handle this differently here.

Reply:
I also was asking myself the same question. I tried removing it but it broke [this test](https://github.com/PostHog/posthog/actions/runs/3230270644/jobs/5288538035) and I just kept it to avoid needing to make even more test changes